oak_restricted_kernel/src/attestation.rs

```diff
@@ -0,0 +1,93 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use crate::{
+    ghcb::GHCB_PROTOCOL, mm::Translator, snp::GUEST_MESSAGE_ENCRYPTOR, ADDRESS_TRANSLATOR,
+    GUEST_HOST_HEAP,
+};
+use alloc::boxed::Box;
+pub use sev_guest::guest::{
+    AttestationReport, AttestationReportData, AuthorKey, EcdsaSignature, GuestPolicy, TcbVersion,
+};
+use sev_guest::guest::{AttestationRequest, AttestationResponse, GuestMessage, ReportStatus};
+use x86_64::VirtAddr;
+
+// The maximum number of custom bytes that can be included in the attestation report.
+const REPORT_DATA_MAX_SIZE: usize = 64;
+
+/// Requests an attestation rerport.
+///
+/// # Arguments
+///
+/// * `report_data` - The custom data that must be included in the report. This is typically used to
+///   bind information (such as the hash of a public key) to the report. A maximum of 64 bytes is
+///   supported.
```

Comment:
I thought it's exactly 64 bytes. I guess it will be padded with zeros if shorter, but I think this function should expect an array of the exact size if possible, WDYT? What does the spec say?

Reply:
I thought a slice might be more flexible, but I agree that the fixed-sized array is clearer. Done.